src/main/java/com/ethanyidong/bunny/Deadline.java

```diff
@@ -0,0 +1,27 @@
+package com.ethanyidong.bunny;
+
+public class Deadline extends Task {
+    private String by;
+
+    public Deadline(String name, String by) {
+        super(name);
+        this.by = by;
+    }
+
+    public String getBy() {
+        return by;
+    }
+
+    public void setBy(String by) {
+        this.by = by;
+    }
+
+    protected String label() {
```

Comment:
It might help to have some comments here explaining what this label is for. Same for some other files.